runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowMetrics.java

```diff
@@ -182,6 +186,13 @@ private Long getCounterValue(MetricUpdate metricUpdate) {
       return ((Number) metricUpdate.getScalar()).longValue();
     }
 
+    private StringSetResult getStringSetValue(MetricUpdate metricUpdate) {
+      if (metricUpdate.getSet() == null) {
+        return StringSetResult.empty();
+      }
+      return StringSetResult.create(new HashSet<>(((ArrayList) metricUpdate.getSet())));
```

Comment:
ImmutableSet.copyOf(...) is likely faster, especially for sets of {0,1} elements. (Might be worth using elsewhere too.)